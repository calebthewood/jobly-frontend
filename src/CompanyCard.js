import { Link } from "react-router-dom";

/** Creates card for each company with a link
 *
 * prop - {company}
 *
 * state - none
 *
 * CompanyList -> CompanyCard
 */

function CompanyCard({ company }) {

  return (
    <Link to={`/companies/${company.handle}`}>
      <div className="row justify-content-center">

        <div className="card shadow mb-4">
          <div className="card-body">
            <h6 className="card-title">
              {company.name}
              {company.logoUrl &&
                <img
                  src={`${company.logoUrl}`}
                  alt={company.handle}
                  className="float-end"
                  style={{ height: "2rem" }} />}
            </h6>
            <p className="card-text fw-light">{company.description}</p>
          </div>
        </div>

      </div>
    </Link>
  );
}

export default CompanyCard;