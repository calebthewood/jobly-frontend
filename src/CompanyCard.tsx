import { Link } from "react-router-dom";
import { ICompanyCardProps } from "./interfaces";

/** Creates card for each company with a link
 *
 * prop - {company}
 *
 * state - none
 *
 * CompanyList -> CompanyCard
 */

function CompanyCard({ company }: ICompanyCardProps) {

  return (
    <Link to={`/companies/${company.handle}`}>
      <div className="row justify-content-center">
        <div className="col-12 col-md-10 col-xl-7">
          <div className="card shadow mb-4">
            <div className="card-body">
              <h6 className="card-title">
                {company.name}
                {company.logoUrl &&
                  <img
                    src={`${company.logoUrl}`}
                    alt={company.handle}
                    className="float-end"
                    style={{ height: "2rem" }} />}
              </h6>
              <p className="card-text fw-light">{company.description}</p>
            </div>
          </div>
        </div>
      </div>
    </Link>
  );
}

export default CompanyCard;