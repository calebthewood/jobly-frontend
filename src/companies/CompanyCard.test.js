import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import CompanyCard from "./CompanyCard";



describe('CompanyCard', () => {
  const companyOne = {
    handle: "rithm",
    name: "Rithm School",
    description: "Become an exceptional developer in 16 weeks.",
    logo_url: "https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"
  };

  const companyTwo = {
    handle: "algo",
    name: "Algo School",
    description: "Become a mediocre developer in 160 weeks."
  };

  it("matches snapshot with logo", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <CompanyCard company={companyOne} />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot without logo", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <CompanyCard company={companyTwo} />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  const mockCompany = {
    handle: 'company-handle',
    name: 'Company Name',
    description: 'Company description',
    logoUrl: 'https://example.com/logo.png',
  };

  it('renders company name and description', () => {
    render(
      <MemoryRouter>
        <CompanyCard company={mockCompany} />
      </MemoryRouter>
    );

    expect(screen.getByText('Company Name')).toBeInTheDocument();
    expect(screen.getByText('Company description')).toBeInTheDocument();
  });

  it('renders company logo if provided', () => {
    render(
      <MemoryRouter>
        <CompanyCard company={mockCompany} />
      </MemoryRouter>
    );

    expect(screen.getByAltText('company-handle')).toBeInTheDocument();
  });

  it('renders a link to the company page', () => {
    render(
      <MemoryRouter>
        <CompanyCard company={mockCompany} />
      </MemoryRouter>
    );

    expect(screen.getByRole('link')).toHaveAttribute(
      'href',
      '/companies/company-handle'
    );
  });
});
