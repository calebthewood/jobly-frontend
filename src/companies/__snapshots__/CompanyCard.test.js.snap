// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyCard matches snapshot with logo 1`] = `
<DocumentFragment>
  <a
    href="/companies/rithm"
    role="link"
  >
    <div
      class="row justify-content-center"
    >
      <div
        class="col-12 col-md-10 col-xl-8"
      >
        <div
          class="card shadow mb-4"
        >
          <div
            class="card-body"
          >
            <h6
              class="card-title"
            >
              Rithm School
            </h6>
            <p
              class="card-text fw-light"
            >
              Become an exceptional developer in 16 weeks.
            </p>
          </div>
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;

exports[`CompanyCard matches snapshot without logo 1`] = `
<DocumentFragment>
  <a
    href="/companies/algo"
    role="link"
  >
    <div
      class="row justify-content-center"
    >
      <div
        class="col-12 col-md-10 col-xl-8"
      >
        <div
          class="card shadow mb-4"
        >
          <div
            class="card-body"
          >
            <h6
              class="card-title"
            >
              Algo School
            </h6>
            <p
              class="card-text fw-light"
            >
              Become a mediocre developer in 160 weeks.
            </p>
          </div>
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
